arrow/formatter.py

```diff
@@ -100,6 +100,10 @@ def _format_token(self, dt, token):
         if token == "X":
             return str(calendar.timegm(dt.utctimetuple()))
 
+        if token == "x":
+            ts = calendar.timegm(dt.utctimetuple()) + (dt.microsecond / 1000000)
+            return str(int(ts * 1000000))
```

Comment:
Do we have to worry about rounding >6 digits of precision here as well?

Reply:
Not here since `datetime` microseconds is capped at 6 digits.

Reply:
Let's put a test in our suite to test this edge case.

Reply:
Ah in that case I don't think a test is necessary then.